include/cantera/numerics/SystemJacobian.h

```diff
@@ -0,0 +1,229 @@
+    //! @f$ \alpha @f$.
+    //! @param rdt  Reciprocal of the time step [1/s]
+    //! @param mask  Mask for transient terms: 1 if transient, 0 if algebraic.
+    virtual void updateTransient(double rdt, int* mask) {
+        throw NotImplementedError("SystemJacobian::updateTransient");
+    }
+
+    //! Solve a linear system using the system matrix *M*
+    //!
+    //! The matrix *M* can either be the preconditioner or the transient Jacobian
+    //! matrix, depending on whether updateTransient() or updatePreconditioner() was
+    //! called last.
+    //!
+    //! @param[in] stateSize length of the rhs and output vectors
+    //! @param[in] rhs_vector right hand side vector used in linear system
+    //! @param[out] output output vector for solution
+    virtual void solve(const size_t stateSize, double* rhs_vector, double* output) {
+        throw NotImplementedError("SystemJacobian::solve");
+    };
+
+    //! Perform preconditioner specific post-reactor setup operations such as factorize.
+    //! @deprecated To be removed after %Cantera 3.2. Use updatePreconditioner() and
+    //!     factorize() instead.
+    virtual void setup() {
+        throw NotImplementedError("SystemJacobian::setup");
+    };
+
+    //! Reset parameters as needed
+    virtual void reset() {
+        throw NotImplementedError("SystemJacobian::reset");
+    };
+
+    //! Called during setup for any processes that need
+    //! to be completed prior to setup functions used in sundials.
+    //! @param networkSize the number of variables in the associated reactor network
+    virtual void initialize(size_t networkSize) {
+        throw NotImplementedError("SystemJacobian::initialize");
+    };
+
+    //! Used to provide system bandwidth for implementations that use banded matrix
+    //! storage. Ignored if not needed.
+    virtual void setBandwidth(size_t bw) {}
+
+    //! Print preconditioner contents
+    virtual void printPreconditioner() {
+        throw NotImplementedError("SystemJacobian::printPreconditioner");
+    };
+
+    //! Set gamma used in preconditioning
+    //! @param gamma used in M = I - gamma*J
+    virtual void setGamma(double gamma) {
+        m_gamma = gamma;
+    };
+
+    //! Get gamma used in preconditioning
+    virtual double gamma() {
+        return m_gamma;
+    };
+
+    //! Set the absolute tolerance in the solver outside of the network initialization
+    //! @param atol the specified tolerance
+    virtual void setAbsoluteTolerance(double atol) {
+        m_atol = atol;
+    }
+
+    //! Get latest status of linear solver. Zero indicates success. Meaning of non-zero
+    //! values is solver dependent.
+    virtual int info() const {
+        throw NotImplementedError("SystemJacobian::info");
+    }
+
+    //! Elapsed CPU time spent computing the Jacobian elements.
+    double elapsedTime() const {
+        return m_elapsed;
+    }
+    void updateElapsed(double evalTime) {
```

Comment:
Missing docstring (here and in several other instances throughout this file).

I realize that this is implicitly covered by what is directly above, but from the perspective of generated documentation (and code generation) this is irrelevant.